yt/data_objects/construction_data_containers.py

```diff
@@ -308,14 +339,20 @@ def get_data(self, fields=None):
         data["pdy"] = self.ds.arr(pdy, code_length)
         data["fields"] = nvals
         # Now we run the finalizer, which is ignored if we don't need it
-        field_data = np.hsplit(data.pop("fields"), len(fields))
+        field_data = np.hsplit(data.pop("fields"), nfields + nsfields)
         for fi, field in enumerate(fields):
             mylog.debug("Setting field %s", field)
             input_units = self._projected_units[field]
-            self[field] = self.ds.arr(field_data[fi].ravel(), input_units)
+            fvals = field_data[fi].ravel()
+            if self.moment == 2:
+                fvals = compute_stddev_image(field_data[fi + nfields].ravel(), fvals)
+            self[field] = self.ds.arr(fvals, input_units)
         for i in list(data.keys()):
             self[i] = data.pop(i)
         mylog.info("Projection completed")
+        if self.moment == 2:
+            for field in sfields:
+                self.ds.field_info.pop(field)
```

Comment:
I'm worried that this isn't necessarily sufficient to completely restore the initial state (in particular in the context of #3948). Unless there's a strong argument to do so, I'd suggest just leaving the field be.

Reply:
We do this elsewhere, see here for the temporary field creation:

https://github.com/yt-project/yt/blob/89073a9b6d91381aff424d977491ef132427456d/yt/visualization/volume_rendering/off_axis_projection.py#L316-L337
and here for its removal:

https://github.com/yt-project/yt/blob/89073a9b6d91381aff424d977491ef132427456d/yt/visualization/volume_rendering/off_axis_projection.py#L402-L403

I'd much prefer a better way of removing fields, but that needs a separate PR. I frankly think it is better to remove it--the `FieldInfoContainer` is pretty jam-packed as is in most situations, and reducing clutter would be ideal.

Reply:
I agree it shouldn't be necessary for this PR, and I wasn't aware we already had examples of this, so thank you for pointing it out. Can you open an issue to remind ourselves that we should better define/encapsulate a method to remove fields ?

Reply:
done here: #4082